SDWebImage/SDWebImageDownloader.m

```diff
@@ -154,16 +156,18 @@ - (void)setValue:(nullable NSString *)value forHTTPHeaderField:(nullable NSStrin
     if (!field) {
         return;
     }
-    NSMutableDictionary *mutableHTTPHeaders = [self.HTTPHeaders mutableCopy];
-    [mutableHTTPHeaders setValue:value forKey:field];
-    self.HTTPHeaders = [mutableHTTPHeaders copy];
+    SD_LOCK(self.HTTPHeadersLock);
+    [self.HTTPHeaders setValue:value forKey:field];
+    SD_UNLOCK(self.HTTPHeadersLock);
 }
 
 - (nullable NSString *)valueForHTTPHeaderField:(nullable NSString *)field {
     if (!field) {
         return nil;
     }
+    SD_LOCK(self.HTTPHeadersLock);
     return [self.HTTPHeaders objectForKey:field];
+    SD_UNLOCK(self.HTTPHeadersLock);
```

Comment:
Forgive me for being stupid, Will  `SD_UNLOCK(self.HTTPHeadersLock)`  be executed ?